File: my-submissions/0205-isomorphic-strings/solution.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isIsomorphic = function (s, t) {
    if (s.length !== t.length) return false;

    let mapS = new Map();
    let mapT = new Map();

    for (let i = 0; i < s.length; i++) {
        const charS = s[i]; // e, g, g
        const charT = t[i]; // a, d, d

        if (mapS.has(charS)) {
            if (mapS.get(charS) !== charT) return false;
        } else {
            mapS.set(charS, charT); // ['e': 'a', 'g' : 'd' ]
        }

        if (mapT.has(charT)) {
            if (mapT.get(charT) !== charS) return false;
        } else {
            mapT.set(charT, charS)
        }
    }

    return true;
};

File: my-submissions/0205-isomorphic-strings/solution.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        
        if (s.length() != t.length()) return false;

        HashMap<Character, Character> mapS = new HashMap<>();
        HashMap<Character, Character> mapT = new HashMap<>();

        for (int i = 0; i < s.length(); i++) {
           char charS = s.charAt(i); // e, g, g
           char charT = t.charAt(i); // a, d, d

            if (mapS.containsKey(charS)) {
                if (mapS.get(charS) != charT) return false;
            } else {
                mapS.put(charS, charT); // ['e': 'a', 'g' : 'd' ]
            }

            if (mapT.containsKey(charT)) {
                if (mapT.get(charT) != charS) return false;
            } else {
                mapT.put(charT, charS);
            }
        }

       return true;
    }
}
